bigtable/client/internal/row_reader.h

```diff
@@ -0,0 +1,158 @@
+// Copyright 2017 Google Inc.
```

Comment:
I think `RowReader` is part of the interface exposed to the user.  They may need to define variables of this type:

```C++
bigtable::RowReader reader = table.ReadRows(...);
```

it cannot be hidden in the `internal` directory or namespace.  We want to be able to say "do not use anything in `internal/` or `bigtable::internal` directly".

Reply:
Thanks for the timely review! I still owe the tests, but replying to the comments:

done - moved